base/uk.ac.stfc.isis.ibex.devicescreens.tests/src/uk/ac/stfc/isis/ibex/devicescreens/tests/DeviceScreenVariablesTest.java

```diff
@@ -0,0 +1,181 @@
+                .thenReturn(getXMLText(expectedName, deviceKey, propertyName, propertyValue));
+        when(mockSwitchableObservable.currentError()).thenReturn(null);
+        when(mockSwitchableObservable.isConnected()).thenReturn(true);
+
+        when(switchingObservableFactory.getSwitchableObservable(any(ChannelType.class),
+                eq(PV_PREFIX + BLOCKSERVER_ADDRESS + GET_SCREENS_SUFFIX))).thenReturn(mockSwitchableObservable);
+        
+        variables = createVariables();
+
+        // Act
+        ForwardingObservable<DeviceScreensDescription> descriptionsObservable = variables.getDeviceScreens();
+
+        // Assert
+        List<DeviceDescription> devices = descriptionsObservable.getValue().getDevices();
+        assertEquals(1, devices.size());
+        assertEquals(expectedName, devices.get(0).getName());
+    }
+
+    @Test
+    public void GIVEN_new_variables_WHEN_get_device_screens_schema_called_THEN_result_is_content_of_schema_pv() {
+        // Arrange
+        String expectedSchema = "This is a test schema";
+        when(mockSwitchableObservable.getValue()).thenReturn(expectedSchema);
+        when(mockSwitchableObservable.currentError()).thenReturn(null);
+        when(mockSwitchableObservable.isConnected()).thenReturn(true);
+        
+        when(switchingObservableFactory.getSwitchableObservable(any(ChannelType.class),
+                eq(PV_PREFIX + BLOCKSERVER_ADDRESS + SCHEMA_SUFFIX)))
+                .thenReturn(mockSwitchableObservable);
+
+        variables = createVariables();
+
+        // Act
+        ForwardingObservable<String> schemaObservable = variables.getDeviceScreensSchema();
+
+        // Assert
+        assertEquals(expectedSchema, schemaObservable.getValue());
+    }
+
+    @Test
+    public void sfsdf() {
+        // Arrange
+        Writable expectedDestination = mock(Writable.class);
+        when(switchingWritableFactory.getSwitchableWritable(any(ChannelType.class),
+                eq(PV_PREFIX + BLOCKSERVER_ADDRESS + SET_SCREENS_SUFFIX))).thenReturn(expectedDestination);
+
+        String deviceName = "device name";
+        String deviceKey = "device key";
+        String propertyKey = "property key";
+        String propertyValue = "property value";
+        DeviceScreensDescription inputValue =
+                getDeviceScreensDescription(deviceName, deviceKey, propertyKey, propertyValue);
+        String expectedConvertedValue = getXMLText(deviceName, deviceKey, propertyKey, propertyValue);
+
+        variables = createVariables();
+
+        // Act
+        Writable<DeviceScreensDescription> setter = variables.getDeviceScreensSetter();
+
+        // Assert
+        verify(expectedDestination, never()).write(any());
+        setter.write(inputValue);
+        verify(expectedDestination, times(1)).write(expectedConvertedValue);
+    }
+
+    private String getXMLText(String deviceName, String deviceKey, String propertyKey, String propertyValue) {
```

Comment:
In the future this should be shared xml function in a test utils class (maybe called mother e.g. DeviceScreensTestDataMother.getXml()) since it is common with other classes. But don't worry about this for this change.